### ISSUE TITLE
Containers list is empty

### ISSUE THREAD

**Issue by djnotes** [containers/podman-desktop#429]
### Bug description

For some reason, containers don't show up in the list, although there are currently several containers running on the system using Podman. I think this happened after I did a *prune* on containers in the terminal. Nevertheless, there are running containers, but none is currently shown. Switched to latest development version. It didn't help.

![image](https://user-images.githubusercontent.com/9320387/187069019-e6065c4d-58ba-4879-b582-28f805860f8c.png)

### Operating system

Fedora 36

### Version

next (development version)

### Steps to reproduce

- Open Podman Desktop 
- Make sure you have a few containers
- Check the containers list

**Comment by benoitf** [comment]
@djnotes could you look at traces displayed in global top menu then View/toggle developper tools and then View console


Also is that View/reload change something ?

**Comment by djnotes** [comment]
Already tried _reload_ to no avail. I'll let you know what happens in dev mode.

**Comment by djnotes** [comment]
So, here's what I see in the Console of the developer mode: 

```
main ↪️ error in engine Podman Error: (HTTP code 500) server error - error getting container from store "30715dc541198c7f9e9e9f0d38374f1919031cf3f67f516fdb308bd83b59b6ba": container not known 
    at /app/lib/io.podman_desktop.PodmanDesktop/resources/app.asar/packages/main/dist/index.js:20:172156
    at c (/app/lib/io.podman_desktop.PodmanDesktop/resources/app.asar/packages/main/dist/index.js:20:172487)
    at Qa.buildPayload (/app/lib/io.podman_desktop.PodmanDesktop/resources/app.asar/packages/main/dist/index.js:20:172128)
    at IncomingMessage.<anonymous> (/app/lib/io.podman_desktop.PodmanDesktop/resources/app.asar/packages/main/dist/index.js:20:171678)
    at IncomingMessage.emit (node:events:539:35)
    at endReadableNT (node:internal/streams/readable:1345:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
﻿


```

Do I need to update my Podman engine?

**Comment by benoitf** [comment]
Hi, maybe restarting the podman machine

**Comment by djnotes** [comment]
Do I have a podman machine on Fedora?

**Comment by benoitf** [comment]
@djnotes no, try the command :  `podman ps -a`

and 

```
curl --unix-socket /run/user/YOUR_ID/podman/podman.sock "http:/v1.41/containers/json?all=true"
```

**Comment by benoitf** [comment]
Issue related to podman issue https://github.com/containers/podman/issues/15526

**Comment by datho7561** [comment]
I ran into this while trying out the podman-desktop flatpak from flathub on RHEL 8.6 with podman 4.1.1. I had no containers or images listed, despite the fact that the "Home" section said that podman was running.

I then tried building podman-desktop from sources and ran the binary executable (it was in the archive `./dist/podman-desktop-0.0.7-next.tar.gz`), and everything appears to work fine (my containers and images are listed).

Maybe there is some difference or limitation to the flatpak version?

**Comment by djnotes** [comment]
@benoitf The result of your curl command returns this:

`{"cause":"container not known","message":"error getting container from store \"30715dc541198c7f9e9e9f0d38374f1919031cf3f67f516fdb308bd83b59b6ba\": container not known","response":500}`

But, podman ps -a returns the list of all containers.

**Comment by benoitf** [comment]
@datho7561 it might be interesting to have the logs on your flatpak installation. I've run successfully Podman Desktop v0.0.6 on Fedora 36


@djnotes yes your error confirmed that the root cause is a bug in the podman engine: https://github.com/containers/podman/issues/15526

**Comment by datho7561** [comment]
@benoitf I reinstalled the flatpak to try to get the logs of the issue, but everything was working fine. Sorry to bother you about that.

**Comment by benoitf** [comment]
@datho7561 no problem, please file any issue you're facing

**Comment by djnotes** [comment]
For me, uninstalling the Flatpak version didn't help. I'm still struggling to see the containers list. I uninstalled the previous version and removed the `.config/Podman Desktop` folder before installing the latest pre-release version (`podman-desktop-0.0.202208310823-85b8232.flatpak`).

**Comment by benoitf** [comment]
@djnotes as I mentioned earlier the bug is on the engine side and not on Desktop side. On macOS/Windows you need to recreate the machine.

On Linux you may try `podman system reset`

**Comment by djnotes** [comment]
Yeah. It's weird. Even, podman system reset didn't fix it.

**Comment by benoitf** [comment]
on my side, podman system reset worked

```
$ podman run --name mariadb --pod new:apps -e MYSQL_RANDOM_ROOT_PASSWORD=yes -d docker.io/library/mariadb:10

$ podman pod rm $(podman pod ls -q)
ERRO[0000] Removing container e97ae3bf7e8295c49efb27ae118153e8fbfa45b0ae3510148725a544a400e86d from pod 89138d4548f4b8ee9ab242f633034a56dec2555b900f8879d68c3b12a3755e64: <nil>
Error: cannot remove container 5a45342d6ac08154bceb2623390a95cc343ede8e796ad3d56b9c148c3cac7bba as it is running - running or paused containers cannot be removed without force: container state improper

$ curl --unix-socket /run/user/1000/podman/podman.sock "http:/v1.41/containers/json?all=true"
{"cause":"container not known","message":"error getting container from store \"e97ae3bf7e8295c49efb27ae118153e8fbfa45b0ae3510148725a544a400e86d\": container not known","response":500}



$ podman system reset
WARNING! This will remove:
        - all containers
        - all pods
        - all images
        - all networks
        - all build cache
        - all machines
Are you sure you want to continue? [y/N] y
ERRO[0002] error freeing lock for container e97ae3bf7e8295c49efb27ae118153e8fbfa45b0ae3510148725a544a400e86d: no such file or directory
ERRO[0002] Removing container e97ae3bf7e8295c49efb27ae118153e8fbfa45b0ae3510148725a544a400e86d from pod 89138d4548f4b8ee9ab242f633034a56dec2555b900f8879d68c3b12a3755e64: error freeing lock for container e97ae3bf7e8295c49efb27ae118153e8fbfa45b0ae3510148725a544a400e86d: no such file or directory
ERRO[0002] Removing Pod 89138d4548f4b8ee9ab242f633034a56dec2555b900f8879d68c3b12a3755e64: error freeing lock for container 5a45342d6ac08154bceb2623390a95cc343ede8e796ad3d56b9c148c3cac7bba: no such file or directory


$ podman system reset
WARNING! This will remove:
        - all containers
        - all pods
        - all images
        - all networks
        - all build cache
        - all machines
Are you sure you want to continue? [y/N] y

$ service podman restart


$ podman system service&

$ curl --unix-socket /run/user/1000/podman/podman.sock "http:/v1.41/containers/json?all=true"
[]
```

**Comment by djnotes** [comment]
Me too, but after rebooting my host computer. Does it make sense?

**Comment by benoitf** [comment]
@djnotes you did the `service podman restart` as well ?

**Comment by djnotes** [comment]
No, just podman system reset

**Comment by djnotes** [comment]
Just FYI, Containers list is not shown on Windows too:

Here's the logs: 

```
main ↪️ error in engine Podman Error: (HTTP code 500) server error - error getting container from store "8c8946fd410672133bb499554bc401b6ad6c395920ed9119b147a9f018b58e2a": container not known 
    at C:\Users\username\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.js:20:172262
    at c (C:\Users\username\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.js:20:172593)
    at ha.buildPayload (C:\Users\username\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.js:20:172234)
    at IncomingMessage.<anonymous> (C:\Users\username\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.js:20:171784)
    at IncomingMessage.emit (node:events:539:35)
    at endReadableNT (node:internal/streams/readable:1345:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
```

I'm using 0.0.202209061634-a1940b7.

**Comment by benoitf** [comment]
@djnotes error is still related to https://github.com/containers/podman/issues/15526

**Comment by benoitf** [comment]
closing as it's fixed in podman 4.3.1
